src/js/worklets/shell/floating_screen.js

```diff
@@ -106,3 +106,51 @@ export function screenBorderRadius(screenState) {
     }
   });
 }
+
+export function screenGestureOnUpdate(screenLeft) {
+  return function (event) {
+    'worklet';
+    const absoluteX = event.absoluteX;
+    if (absoluteX !== null) {
+      screenLeft.value = event.absoluteX;
+    }
+  };
+}
+
+export function screenGestureOnEnd(
+  screenLeft,
+  screenState,
+  screenWidth,
+  leftVelocity,
+  rightVelocity,
+  screenClosedCallback,
+) {
+  return function (event) {
+    'worklet';
+    const absoluteX = event.absoluteX ?? 0;
+    const velocityX = event.velocityX ?? 0;
+    var closeScreen = false;
+    if (velocityX > rightVelocity || (velocityX > leftVelocity && absoluteX >= screenWidth / 2)) {
+      closeScreen = true;
+    }
+
+    // Velocity (points/sec) = Distance/time
+    var animationVelocity = (screenWidth * 1000) / constants.SHELL_ANIMATION_TIME;
+
+    if (Math.abs(velocityX) > animationVelocity) {
+      animationVelocity = velocityX; // Faster fling
+    }
+
+    const newDistance = closeScreen ? screenWidth - absoluteX : absoluteX;
+    const animationTime = (newDistance * 1000) / animationVelocity;
+
+    screenLeft.value = withTiming(closeScreen ? screenWidth : 0, {
```

Comment:
is it not possible to return the value and set this value where it is being used. In general we don't want to mutate args with FP.

Reply:
then we will lose benefits of worklets. This is recommended way. [Example](https://docs.swmansion.com/react-native-reanimated/docs/api/hooks/useAnimatedGestureHandler)

Reply:
ah okay, thanks for the link!